channeldb/migrations_test.go

```diff
@@ -5,15 +5,439 @@ import (
+			}
+
+			// Then we delete or modify the information in the
+			// database according to the testcase.
+			err := db.Update(func(tx *bbolt.Tx) error {
+
+				// Set the edge policies.
+				err := setPolicy(
+					tx, info.NodeKey1Bytes,
+					policy1.ChannelID, test.policy1Status,
+				)
+				if err != nil {
+					return err
+				}
+
+				err = setPolicy(
+					tx, info.NodeKey2Bytes,
+					policy2.ChannelID, test.policy2Status,
+				)
+				if err != nil {
+					return err
+				}
+
+				// Set the nodes.
+				err = setNode(
+					tx, info.NodeKey1Bytes,
+					test.node1Exists,
+				)
+				if err != nil {
+					return err
+				}
+
+				err = setNode(
+					tx, info.NodeKey2Bytes,
+					test.node2Exists,
+				)
+				if err != nil {
+					return err
+				}
+
+				// And finally set the edge info.
+				err = setInfo(tx, info, test.edgeInfoExists)
+				if err != nil {
+					return err
+				}
+
+				return nil
+			})
+			if err != nil {
+				t.Fatalf("unable to update db: %v", err)
+			}
+
+			chanID = info.ChannelID
+		}
+
+		// afterMigrationFunc asserts that the db is migrated to the
+		// expected format.
+		afterMigrationFunc := func(db *DB) {
+			meta, err := db.FetchMeta(nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if meta.DbVersionNumber != 1 {
+				t.Fatal("migration 'migrateEdgePolicies' " +
+					"wasn't applied")
+			}
+
+			graph := db.ChannelGraph()
+
+			// If we were migrating from a DB having the edge info
+			// present, it SHOULD still be present. If the original
+			// DB didn't have the edge info, it should not be
+			// there.
+			shouldExist := test.edgeInfoExists
+			_, _, found, err := graph.HasChannelEdge(chanID)
+			if err != nil {
+				t.Fatalf("unable to query for edge: %v", err)
+			}
+
+			if shouldExist != found {
+				t.Fatalf("expected to find edge: %v ",
+					test.edgeInfoExists)
+			}
+
+			// If the edge existed, we should be able to properly
+			// fetch policies and nodes.
+			if shouldExist {
+				e, _, _, err := graph.FetchChannelEdgesByID(
+					chanID,
+				)
+				if err != nil {
+					t.Fatalf("unable to fetch edge: %v",
+						err)
+				}
+
+				if e == nil {
+					t.Fatalf("expected edge info to be " +
+						"present")
+				}
+
+				err = db.View(func(tx *bbolt.Tx) error {
+					err := assertMigratedPolicy(
+						tx, e.NodeKey1Bytes,
+						e.ChannelID, test.policy1Status,
+					)
+					if err != nil {
+						return err
+					}
+					return assertMigratedPolicy(
+						tx, e.NodeKey2Bytes,
+						e.ChannelID, test.policy2Status,
+					)
+
+				})
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				// Both nodes should be there.
+				var numNodes int
+				err = graph.ForEachNode(nil,
+					func(*bbolt.Tx, *LightningNode) error {
+						numNodes++
+						return nil
+					})
+				if err != nil {
+					t.Fatalf("unable to terate nodes: %v",
+						err)
+				}
+
+				if numNodes != 2 {
+					t.Fatalf("expected 2 nodes, found %v",
+						numNodes)
+				}
+			}
```

Comment:
Good idea, but I think that would increase the scope of the migration. Currently it checks all `edgeinfos` and add corresponding policies if they are missing. 

In this case we would need to iterate over all policies, and delete any where we cannot find the info. Do you think this is worthwhile to add?

Reply:
I am just thinking that we now do a "fix" migration that still leaves some undesired state in the database. If we don't do it now, it may need another fix migration and db version number later.

Reply:
Ok, added the cleanup 👍 

Note that still will only happen for users with this db version. Users already past this migration won't get this cleanup.

Reply:
I have a node that already upgraded the db to the latest version, but still has a bad policy requiring a strategic continue. This pr is not going to help me there. I was wondering, shouldn't this cleanup be a new db version so that it is beneficial for everyone?